# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/wr/webrender/src/render_task.rs
# Commit: 15da94c6a7b1
# Full Hash: 15da94c6a7b19f971d3dffaa4cab8edb8e41cb62
# Author: Glenn Watson <git@intuitionlibrary.com>
# Date: 2020-12-15 09:29:54
# Regressor Bug: 1682365
# File Overlap Count: 1
# Description:
#   Bug 1682365 - Pt 2 - Remove texture array usage from render task graph. r=nical
#   
#   With this change, all color/alpha intermediate surfaces are individual
#   textures, rather than texture arrays.
#   
# ==============================================================================

diff -r 72ddac486037 -r 15da94c6a7b1 gfx/wr/webrender/src/render_task.rs
--- a/gfx/wr/webrender/src/render_task.rs	Tue Dec 15 01:46:51 2020 +0200
+++ b/gfx/wr/webrender/src/render_task.rs	Mon Dec 14 21:48:41 2020 +0000
@@ -64,7 +64,7 @@
     /// build phase, we invoke `RenderTargetList::alloc()` and store the
     /// resulting location in the first member. That location identifies the
     /// render target and the offset of the allocated region within that target.
-    Dynamic(Option<(DeviceIntPoint, CacheTextureId, RenderTargetIndex)>, DeviceIntSize),
+    Dynamic(Option<(DeviceIntPoint, CacheTextureId)>, DeviceIntSize),
     /// The output of the `RenderTask` will be persisted beyond this frame, and
     /// thus should be drawn into the `TextureCache`.
     TextureCache {
@@ -106,7 +106,7 @@
     pub fn to_source_rect(&self) -> (DeviceIntRect, LayerIndex) {
         match *self {
             RenderTaskLocation::Dynamic(None, _) => panic!("Expected position to be set for the task!"),
-            RenderTaskLocation::Dynamic(Some((origin, _, layer)), size) => (DeviceIntRect::new(origin, size), layer.0 as LayerIndex),
+            RenderTaskLocation::Dynamic(Some((origin, _)), size) => (DeviceIntRect::new(origin, size), 0),
             RenderTaskLocation::TextureCache { rect, layer, .. } => (rect, layer),
             RenderTaskLocation::PictureCache { .. } => {
                 panic!("bug: picture cache tasks should never be a source!");
@@ -1340,7 +1340,7 @@
 
     pub fn get_target_texture(&self) -> CacheTextureId {
         match self.location {
-            RenderTaskLocation::Dynamic(Some((_, texture_id, _)), _) => {
+            RenderTaskLocation::Dynamic(Some((_, texture_id)), _) => {
                 assert_ne!(texture_id, CacheTextureId::INVALID);
                 texture_id
             }
@@ -1368,8 +1368,8 @@
             // TODO(gw): Consider some kind of tag or other method
             //           to mark a task as unused explicitly. This
             //           would allow us to restore this debug check.
-            RenderTaskLocation::Dynamic(Some((origin, _, target_index)), size) => {
-                (DeviceIntRect::new(origin, size), target_index)
+            RenderTaskLocation::Dynamic(Some((origin, _)), size) => {
+                (DeviceIntRect::new(origin, size), RenderTargetIndex(0))
             }
             RenderTaskLocation::Dynamic(None, _) => {
                 (DeviceIntRect::zero(), RenderTargetIndex(0))